### ISSUE TITLE
Deploy chalice app to HTTP APIs

### ISSUE THREAD

**Issue by danil-smirnov** [aws/chalice#1321]
AWS has recently announced [API Gateway HTTP APIs](https://aws.amazon.com/ru/about-aws/whats-new/2019/12/amazon-api-gateway-offers-faster-cheaper-simpler-apis-using-http-apis-preview/), which are _up to 71% cheaper compared to REST APIs_.

I suppose all and everyone are now keen to deploy their chalice apps to this exciting new infra!

**Comment by benjaminsanborn** [comment]
any perspective on if the chalice project is interested in targeting HTTP Gateway?

**Comment by jamesls** [comment]
Yes!  Just need to work out the API we want for being able to configure this.  I also haven't had much chance to play around with these new API type so I'm not sure what's involved in supporting this.

**Comment by danil-smirnov** [comment]
@jamesls Any estimation on this? I believe this feature should be prioritised as it allows Chalice users to save costs on API Gateway service...

**Comment by jamesls** [comment]
Let me dig into this a little more so I can get a better idea of what's needed to support this.

**Comment by JPVan** [comment]
@jamesls Excited for this. :)

**Comment by kapilt** [comment]
isn't it just another configuration on API Gateway v2 which chalice already supports as part of web sockets? wrt to decorator expression, maybe @app.http_route or @app.route2() just spitballing to start a conversation ;-)

**Comment by ricky-sb** [comment]
Do we even need a separate decorator? This could be as easy as a boolean in config.json to deploy to HTTP APIs vs. REST APIs.

**Comment by subssn21** [comment]
They just added support for lambda and iam authorizers for http apis. I don’t think you would want to make it a boolean on the current decorator since they have different feature sets.

**Comment by jamesls** [comment]
Yeah, if we do go down the route of separate APIs I'd imagine they'd be as similar as possible so there's minimal adjustments needed to switch/migrate over.  From what I can tell, HTTP APIs are different enough that I'm not sure we'd be able to share APIs.

**Comment by jdalzatec** [comment]
Hi guys, anyone knows what is the status of this feature? Thanks in advance.

**Comment by bl-ue** [comment]
🏓 @jamesls — any ETA?  We need this :)

**Comment by Glocktober** [comment]
Adding to this request...
I've been working with these for a while with straight lambda functions, but it would be useful to capitalize on the Chalice framework for some projects.

**Comment by baotran2207** [comment]
Any update for this feature ? :(

**Comment by rajeshkatkarnice** [comment]
any updates?

**Comment by softwareguy74** [comment]
Any updates?  Can't use it until this is implemented.

**Comment by baotran2207** [comment]
+1 this feature

**Comment by sladg** [comment]
+1 for this, are there any microservice packages for HttpApi available?

**Comment by matteocontrini** [comment]
@sladg take a look at AWS Lambda Powertools for Python

**Comment by kapilt** [comment]
https://github.com/aws/chalice/issues/2067
